### PR TITLE
make the usage function consistent across scripts

### PR DESCRIPTION
**Pull request by rafasc**
Make the usage function print the usage string to stderr and
exit with error.

### CODE REVIEW

File: lib/aur-vercmp [hunk 1]
```diff
@@ -68,7 +68,7 @@ trap_exit() {
 }
 
 usage() {
-    plain "usage: $argv0 [-p path] [-acq]" >&2
+    plain 'usage: %s [-p path] [-acq]' "$argv0" >&2
```
Comment (AladW) [inline lib/aur-vercmp]: I guess if we're aiming for consistency and add `%s` everywhere (less shellcheck complaints?), the same should be done for `error`/`warning`

File: lib/aur-search [hunk 2]
```diff
@@ -106,7 +106,7 @@ trap_exit() {
 }
 
 usage() {
-    printf >&2 'usage: %s: [-adimnqrsv] [-k key] pkgname...\n' "$argv0"
+    plain 'usage: %s [-adimnqrsv] [-k key] pkgname...' "$argv0" >&2
```
Comment (AladW) [inline lib/aur-search]: Maybe add a comment in the relevant `[[ -t 1 ]]` section too (re. #473, as a reminder to fix it in a different PR)

File: lib/aur-vercmp (hunk 1 above)
Comment (rafasc) [reply lib/aur-vercmp]: > I guess if we're aiming for consistency and add %s everywhere (less shellcheck complaints?)

I don't think shellcheck complains about having the format:
```diff
-    plain "usage: $argv0 [-p path] [-acq]" >&2
```

I just picked one and went with it. If you prefer the this, I can easily change all to this format. I only care about being consistent across the board.

> the same should be done for error/warning
Agreed, The original intent for this PR was to make `usage()` consistent. So I restricted to changes to usage.

We can rename this PR something else and add commits fixing there rest.

File: lib/aur-search (hunk 2 above)
Comment (rafasc) [reply lib/aur-search]: Out of scope here. I'll open an issue to keep track of it.

File: lib/aur-vercmp (hunk 1 above)
Comment (AladW) [reply lib/aur-vercmp]: Works for me as is. We can make the changes to plain et al in a second PR.

**Comment by AladW** [comment]
Self-requested, really... :roll_eyes:

**Pull request merged**